### PR TITLE
Support for MultiPolygon and MultiLineString

### PR DESCRIPTION
**Pull request by jacobo**
We needed support for this and it seems relatively non-invasive to add. This works for our specific use case, but I only tested it by directly editing the released `ol3-gm-debug.js`

I'd love to see this support included in the next release so we don't have to maintain a fork or patch, please let me know what I can do next to help this along

### CODE REVIEW

**Comment by adube** [comment]
Thanks a lot for the contribution.  I should be able to review this soon (later this week).  You can poke me if I forget.

**Comment by adube** [comment]
@jacobo It would be nice if an example in which we demonstrate the support of multi polygons was added.  We could call it "multi-geometries" and only demonstrate polygons (for now).

Would you please add this as part of this PR?  Thanks.

**Comment by adube** [comment]
@jacobo (in case you missed it) See my last comment.  Thanks.

**Comment by jacobo** [comment]
thanks, haven't forgotten, currently on paternity leave so this may take a while 👶

**Comment by adube** [comment]
No problem.  Congratulations, btw :)

**Comment by LSG8** [comment]
How to add multilinestring too? I had a look on @jacobo 's code for multipolygon and tried similar things for multilinestring. But it didn't work. I have attached here my modified gm.js as a .txt file
[modified_gm_for_multi_line_n_polygon.js.txt](https://github.com/mapgears/ol3-google-maps/files/1470710/modified_gm_for_multi_line_n_polygon.js.txt)
. Please help me out.

**Comment by adube** [comment]
@LSG8 Please, provide a separate PR.  This would help seeing what was changed.

**Comment by jacobo** [comment]
screen shot 2017-11-29 at 10 35 58 am

I have added an example for multi-polygon to concept.js ( a pinwheel over Ontario and some hand-drawn polygons over the great lakes)

The key difference between MultiPolygon and just using multiple Polygon objects is that it makes a single feature that can be dragged around the map as one.

**Comment by adube** [comment]
That's an excellent idea, i.e. it's better to show this there instead of yet an other example.  Thanks.

I haven't checked: is your code rebased onto master?

**Comment by adube** [comment]
Nevermind, it's an obvious "yes".  Alright I'll review this PR this week.

File: src/olgm/gm.js [hunk 1]
```diff
@@ -93,26 +95,31 @@ olgm.gm.createGeometry = function(geometry, opt_ol3map) {
...
+    return latLngs;
+  };
 
   var gmapGeometry = null;
-  if (geometry instanceof ol.geom.LineString) {
-    gmapGeometry = new google.maps.Data.LineString(latLngs);
+
+  if (geometry instanceof ol.geom.MultiPolygon) {
+    var latLngsArr = [];
+    geometry.getCoordinates()[0].forEach(function(coordsArr) {
```
Comment (adube) [inline src/olgm/gm.js]: Instead of a `forEach`, let's use a standard `for` loop as it is faster.

File: src/olgm/gm.js [hunk 2]
```diff
@@ -93,26 +95,31 @@ olgm.gm.createGeometry = function(geometry, opt_ol3map) {
...
+      latLngs.push(new google.maps.LatLng(lonLatCoords[1], lonLatCoords[0]));
+    }
+    return latLngs;
+  };
 
   var gmapGeometry = null;
-  if (geometry instanceof ol.geom.LineString) {
-    gmapGeometry = new google.maps.Data.LineString(latLngs);
+
+  if (geometry instanceof ol.geom.MultiPolygon) {
```
Comment (adube) [inline src/olgm/gm.js]: Let's move this at the end, i.e.:

1: if line string
2: else if polygons
3: else (multi polygons)

**Review by adube — CHANGES_REQUESTED** [review]
Very minor things to change, then this would be good to merge.  Thanks a lot for the contribution.

**Comment by LSG8** [comment]
@jacobo Can you help me in PR #194 ?

**Comment by LSG8** [comment]
@jacobo Can you help me in PR #194 ? I modified the gm.js for multilinestring similar to your modification for multipolygon. But it is not working. I don't get any at the time of `make dist`. But no multilinestring.

**Comment by LSG8** [comment]
@jacobo Can you help me in PR #194 ? I modified the gm.js for multilinestring similar to your modification for multipolygon. But it is not working. I don't get any error at the time of `make dist`. But no multilinestring.

**Comment by jacobo** [comment]
screen shot 2017-11-30 at 3 50 42 pm

multiline string example looks like this ^

**Comment by jacobo** [comment]
@LSG8 does my implementation of multi line string work for you?

**Comment by jacobo** [comment]
@adube thanks for your responsiveness, hopefully you don't mind my conflation of 2 features in one PR...

**Comment by adube** [comment]
@jacobo I don't mind :)  I'll review this today. Thank you very much.

**Review by adube — APPROVED** [review]
Very nice!  Code is clean and the addition to the example made me smile.

**Comment by LSG8** [comment]
Thanks @jacobo it is working.

**Comment by jacobo** [comment]
thanks for the merge @adube any plans to release soon? (I'd like to consume via npm)

**Comment by adube** [comment]
We could make an other upgrade of OL first, like we've been doing the fast few weeks.  Once done, it would make sense to make a new release.

@bampakoa Are you planning on working for a new upgrade to a new version of OL soon (this week)?  If so, I'd gladly review.  Thanks.

**Comment by bampakoa** [comment]
@adube I plan to submit a PR this week concerning OL version 4.4.2

**Comment by adube** [comment]
Excellent, thanks.

**Comment by adube** [comment]
@jacobo 0.17.0 released: https://www.npmjs.com/package/ol3-google-maps

**Comment by jacobo** [comment]
Awesome thanks @adube I can confirm it's working!

**Pull request merged**